src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -923,43 +933,40 @@ public RequestLineStatus TakeStartLine(SocketInput input)
 
                 if (pathBegin.Peek() == ' ')
                 {
-                    RejectRequest(RequestRejectionReason.MissingRequestTarget);
+                    RejectRequest(RequestRejectionReason.InvalidRequestLine,
+                        Log.IsEnabled(LogLevel.Information) ? start.GetAsciiStringEscaped(end, MaxInvalidRequestLineChars) : string.Empty);
                 }
 
                 scan.Take();
                 begin = scan;
                 if (scan.Seek(ref _vectorCRs, ref end) == -1)
                 {
-                    RejectRequest(RequestRejectionReason.MissingCrAfterVersion);
+                    RejectRequest(RequestRejectionReason.InvalidRequestLine,
+                        Log.IsEnabled(LogLevel.Information) ? start.GetAsciiStringEscaped(end, MaxInvalidRequestLineChars) : string.Empty);
                 }
 
                 string httpVersion;
                 if (!begin.GetKnownVersion(out httpVersion))
                 {
-                    // A slower fallback is necessary since the iterator's PeekLong() method
-                    // used in GetKnownVersion() only examines two memory blocks at most.
-                    // Although unlikely, it is possible that the 8 bytes forming the version
-                    // could be spread out on more than two blocks, if the connection
-                    // happens to be unusually slow.
-                    httpVersion = begin.GetAsciiString(scan);
-
-                    if (httpVersion == null)
+                    httpVersion = begin.GetAsciiStringEscaped(scan, 8);
+
+                    if (httpVersion == string.Empty)
                     {
-                        RejectRequest(RequestRejectionReason.MissingHTTPVersion);
+                        RejectRequest(RequestRejectionReason.InvalidRequestLine,
+                            Log.IsEnabled(LogLevel.Information) ? start.GetAsciiStringEscaped(end, MaxInvalidRequestLineChars) : string.Empty);
                     }
-                    else if (httpVersion != "HTTP/1.0" && httpVersion != "HTTP/1.1")
+                    else
                     {
-                        RejectRequest(RequestRejectionReason.UnrecognizedHTTPVersion);
+                        RejectRequest(RequestRejectionReason.UnrecognizedHTTPVersion, httpVersion);
                     }
                 }
 
                 scan.Take(); // consume CR
-                if (scan.Block != end.Block || scan.Index != end.Index)
+                if (scan.Take() != '\n')
                 {
-                    RejectRequest(RequestRejectionReason.MissingLFInRequestLine);
+                    RejectRequest(RequestRejectionReason.InvalidRequestLine,
+                        Log.IsEnabled(LogLevel.Information) ? start.GetAsciiStringEscaped(end, MaxInvalidRequestLineChars) : string.Empty);
                 }
-                scan.Take(); // consume LF
-                end = scan;
```

Comment:
It doesn't look the end is being advanced passed the line feed any more. Doesn't this lead to a bad examined value in the call to ConsumingComplete?